lib/FusionInventory/Agent/Task/WakeOnLan.pm

```diff
@@ -151,23 +153,37 @@ sub _getInterface {
             @interfaces = FusionInventory::Agent::Tools::Win32::getInterfaces(
                 logger => $self->{logger}
             );
+            # on Windows, we have to use internal device name instead of litteral name
+            for my $interface ( @interfaces ) {
+                $interface->{DESCRIPTION} =
+                    $self->_getWin32InterfaceId($interface->{PNPDEVICEID})
+            }
+
             last;
         }
     }
 
+    my @nonloopbackordumbinterfaces =
+        grep { $_->{DESCRIPTION} ne 'lo' }
+        grep { $_->{IPADDRESS} }
+        grep { $_->{MACADDR} }
+        @interfaces;
+
+   return @nonloopbackordumbinterfaces;
+}
+
+sub _getInterface {
```

Comment:
I was unsure about "private" and "really unused elsewhere" and potential need about backward compatibility. I'll try to remove _getInterface() and update my PR now.

Flooding WOL packets could hurt in some edge cases : on my network, a WOL packet to whatever mac address trigger 2 replies from 2 prints that says "Unsupported command". Yes they listen on UDP 9 discard port, they accept broadcasts and they reply !

Regards,